### PR TITLE
[SYCL] Add test for std::byte

### PR DESCRIPTION
**Pull request by maximdimakov**
This test is for https://github.com/intel/llvm/pull/4705
Signed-off-by: mdimakov <maxim.dimakov@intel.com>

### CODE REVIEW

**Review by s-kanaev — DISMISSED** [review]
Seems legit.

File: SYCL/Basic/group_async_copy.cpp [hunk 1]
```diff
@@ -24,14 +28,14 @@ template <typename T> void initInputBuffer(buffer<T, 1> &Buf, size_t Stride) {
   auto Acc = Buf.template get_access<access::mode::write>();
   for (size_t I = 0; I < Buf.get_count(); I += WorkGroupSize) {
     for (size_t J = 0; J < WorkGroupSize; J++)
-      Acc[I + J] = I + J + ((J % Stride == 0) ? 100 : 0);
+      Acc[I + J] = (T)(I + J + ((J % Stride == 0) ? 100 : 0));
```
Comment (alexbatashev) [inline SYCL/Basic/group_async_copy.cpp]: Is there any reason for using C-style cast?

File: SYCL/Basic/group_async_copy.cpp [hunk 2]
```diff
@@ -67,7 +71,16 @@ template <typename T> std::string toString(vec<T, 4> A) {
 }
 
 template <typename T = void>
-typename std::enable_if<!is_vec<T>::value, std::string>::type toString(T A) {
+typename std::enable_if<!is_vec<T>::value && std::is_same<T, std::byte>::value,
+                        std::string>::type
```
Comment (alexbatashev) [inline SYCL/Basic/group_async_copy.cpp]:
```suggestion
std::enable_if_t<!is_vec<T>::value && std::is_same<T, std::byte>::value,
                        std::string>
```

File: SYCL/Basic/group_async_copy.cpp (hunk 1 above)
Comment (maximdimakov) [reply SYCL/Basic/group_async_copy.cpp]: I replaced it with static_cast

File: SYCL/Basic/group_async_copy.cpp (hunk 2 above)
Comment (maximdimakov) [reply SYCL/Basic/group_async_copy.cpp]: Replaced

**Review by s-kanaev — APPROVED** [review]
LGTM

**Pull request merged**